cmd/lnmuxd/run.go

```diff
@@ -48,6 +48,17 @@ func runAction(c *cli.Context) error {
 		return err
 	}
 
+	group, ctx := errgroup.WithContext(context.Background())
+
+	group.Go(func() error {
```

Comment:
I still don't fully understand why this needs to run in a goroutine. Wouldn't something like this work?

```golang
err := run(ctx, cfg, group)
if err != nil {
   cancel()

  // Error can be ignored because we already have the primary error
   _ = group.Wait()

  return err
}

return group.Wait()
```

Maybe it comes down to the same thing though...

Reply:
Just for this refactor commit, I think the extra goroutine isn't necessary yet? It would be sufficient to just `group.Wait` if err == nil, in line with the original code?

Reply:
Yes indeed it works this way, it was my first approach.

But using the goroutine permits to avoid the double `group.Wait()` (I know it is in different paths but.. still).

But if you think this is more readable, I will go with this approach!

Reply:
It would be my preference to avoid the nested goroutine, for readability